packages/hash/frontend/src/pages/[account-slug]/types/entity-type/entity-type-selector.tsx

```diff
@@ -0,0 +1,232 @@
+import { EntityType } from "@blockprotocol/type-system-web";
+import { faSearch } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon, TextField } from "@hashintel/hash-design-system";
+import {
+  Autocomplete,
+  Box,
+  outlinedInputClasses,
+  PopperProps,
+  Typography,
+} from "@mui/material";
+import { FunctionComponent, useMemo, useRef, useState } from "react";
+import { ArrowUpRightIcon } from "../../../../shared/icons/svg";
+import {
+  OntologyChip,
+  parseUriForOntologyChip,
+} from "../../shared/ontology-chip";
+import {
+  TypeListSelectorDropdown,
+  TypeListSelectorDropdownContext,
+  TYPE_SELECTOR_HEIGHT,
+} from "./type-list-selector-dropdown";
+import { useEntityTypes } from "./use-entity-types";
+
+export const EntityTypeSelector: FunctionComponent<{
+  onSelect: (entityType: EntityType) => void;
+  onCancel: () => void;
+  onCreateNew: (name: string) => void;
+}> = ({ onCancel, onSelect, onCreateNew }) => {
+  const [search, setSearch] = useState("");
+  const entityTypesObject = useEntityTypes();
+  const entityTypes = Object.values(entityTypesObject ?? {});
+
+  const modifiers = useMemo(
+    (): PopperProps["modifiers"] => [
+      {
+        name: "addPositionClass",
+        enabled: true,
+        phase: "write",
+        fn({ state }) {
+          if (state.elements.reference instanceof HTMLElement) {
+            state.elements.reference.setAttribute(
+              "data-popper-placement",
+              state.placement,
+            );
+          }
+        },
+      },
+      {
+        name: "preventOverflow",
+        enabled: false,
+      },
+    ],
+    [],
+  );
+
+  const [open, setOpen] = useState(false);
+  const highlightedRef = useRef<null | EntityType>(null);
+
+  return (
+    <TypeListSelectorDropdownContext.Provider
+      // eslint-disable-next-line react/jsx-no-constructed-context-values
+      value={{
+        query: search,
+        createButtonProps: {
+          onMouseDown: (evt) => {
+            evt.preventDefault();
+            evt.stopPropagation();
+            onCreateNew(search);
+          },
+        },
+        variant: "entityType",
+      }}
+    >
+      <Autocomplete
```

Comment:
Can we add a `@todo` to look at reusing some of this to stop repetition here and in the entity type editor property type selector?

Reply:
Addressed here https://github.com/hashintel/hash/pull/1492/commits/d83d521d7167144f024f073e894990567dfd4ed0